presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentControlUtils.java

```diff
@@ -0,0 +1,220 @@
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).entity("Presto is not installed").build();
+        }
+        new Thread(() -> {
+            try {
+                int startPresto = executeCommand(120, "service", "presto", "startPresto");
+                if (startPresto != 0) {
+                    LOGGER.error("Failed to startPresto presto");
+                }
+            }
+            catch (PrestoManagerException e) {
+                LOGGER.error(e.getMessage(), e.getCause());
+            }
+            catch (Exception e) {
+                LOGGER.error("Unhandled exception", e);
+            }
+        }).start();
+        return Response.status(ACCEPTED).entity("Check status to make sure that presto has started").build();
+    }
+
+    public static Response stopPresto(StopType stopType)
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).build();
+        }
+        if (!isRunning()) {
+            LOGGER.debug("Presto is not running");
+            return Response.status(OK).build();
+        }
+        try {
+            stopHelper(stopType);
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    private static void stopHelper(StopType stopType)
+            throws PrestoManagerException
+    {
+        switch (stopType) {
+            case TERMINATE:
+                int prestoTerminate = executeCommand("service", "presto", "stop");
+                if (prestoTerminate != 0) {
+                    throw new PrestoManagerException("Failed to stop presto", prestoTerminate);
+                }
+                break;
+            case KILL:
+                executeCommand("sudo", LAUNCHER_SCRIPT, "kill");
+                int prestoKill = executeCommand("service", "presto", "status");
+                if (prestoKill != 3) {
+                    throw new PrestoManagerException("Failed to stop presto", prestoKill);
+                }
+                break;
+            case GRACEFUL:
+                try {
+                    if (getFileProperty(PATH_TO_CONFIG_FILE, "coordinator").equalsIgnoreCase("true")) {
+                        throw new PrestoManagerException("Coordinator can't be gracefully stopped");
+                    }
+                    String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+                    Response response = newRequestBuilder(new URI(format("localhost:%s/v1/info/state", prestoPort)))
+                            .withBody(entity((new Gson().toJson("SHUTTING_DOWN")), APPLICATION_JSON_TYPE)).buildPut().invoke();
```

Comment:
This `Gson` object is used in multiple places, create a private field so that you could use only one. I don't feel strong about this though.

However, if the only message is a String, maybe you could just put it as Text/plain